src/state.py

```diff
@@ -153,11 +156,33 @@ def _fetch_agent_password(model: ops.Model, config: WazuhConfig) -> str | None:
     return agent_password_content
 
 
+def _fetch_cluster_key(model: ops.Model) -> str:
+    """Fetch the Wazuh cluster key.
+
+    Args:
+        model: the Juju model.
+
+    Returns: the key for the cluster, if any.
+
+    Raises:
+        InvalidStateError: if the secret when the key should reside is invalid.
+    """
+    try:
+        cluster_key_secret = model.get_secret(label=WAZUH_CLUSTER_KEY_SECRET_LABEL)
+    except ops.SecretNotFoundError as exc:
+        raise InvalidStateError("Cluster key secret.") from exc
+    cluster_key_content = cluster_key_secret.get_content(refresh=True).get("value")
+    if not cluster_key_content:
+        raise InvalidStateError("Cluster key secret does not contain the expected key 'value'.")
```

Comment:
In that case we should probably ensure the charm gets into an error state, there's not much a user can do in such situation unless remove/redeploy

Reply:
Or regenerate the secret as previously suggested

Reply:
Invalid leads to Blocked state, which I think is the appropriate thing to do here since the error is recoverable only if the operator either redeploys or manually creates/fixes the secret

Reply:
How about consistency? We are setting the charm in blocked state when other secrets are invalid (either operator defined or sent via relation data)

Reply:
When the secret comes from relation the charm shouldn't be blocked because the user can't do much.
Blocked vs Error should be fairly simple:
- Do I expect the user to take an action to fix this issue ? Is it well described in the BlockedStatus message or documented that this is expected (like allowed values in a config) ? If yes to both, then put in Blocked
- Otherwise Error

In that case:
Do I expect the user to take an action to fix this issue ?
A: Not sure, in most of the case it's automatically set, if that's not set do we want the user to take the burden of creating the secret, setting a random value, grant it to the application and label it while it should be automatic ?
Is it well described in the BlockedStatus message or documented that this is expected (like allowed values in a config) ?
A: No

So if we want blocked state, we need a good reasoning on why something that should be transparent to the user becomes not transparent. We should state exactly to the user what's expected from him, and document how to achieve it and in which circumstances

I think we don't want to bother too much with the documentation, if the secret is not generated it's likely a transient issue. ErrorState, and redeploy is probably the easiest approach.
The best approach is to generate the secret whenever it is missing, so that the user never have to know and bother about it

Reply:
I'll review all the BlockedStates accordingly

Reply:
Updated